ui/nuxeo-filters-behavior.js

```diff
@@ -62,6 +63,32 @@ export const FiltersBehavior = {
     return false;
   },
 
+  canSetRetention(document) {
+    return (
+      document &&
+      !document.hasLegalHold &&
+      this.hasPermission(document, 'MakeRecord') &&
+      this.hasPermission(document, 'SetRetention') &&
+      Object.prototype.hasOwnProperty.call(document.properties, 'file:content')
+    );
+  },
+
+  canSetLegalHold(document) {
+    return (
+      document &&
```

Comment:
Checking `document` here is not needed, since the method `hasPermission` already checks it.